frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/NodeVersionDetector.java

```diff
@@ -0,0 +1,160 @@
+package com.github.eirslett.maven.plugins.frontend.lib;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Objects.isNull;
+import static java.util.Optional.empty;
+import static org.slf4j.LoggerFactory.getLogger;
+
+public class NodeVersionDetector {
+
+    private static final String TOOL_VERSIONS_FILENAME = ".tool-versions";
```

Comment:
do we care about mise.toml? https://sourcegraph-frontend.prod.services.kitt-inf.net/bitbucket.org/atlassian/dc-classifications/-/blob/.mise.toml

Reply:
Excellent point, it's cheap to add just in case